win32ss/user/ntuser/menu.c

```diff
@@ -3282,11 +3282,17 @@ static void FASTCALL MENU_HideSubPopups(PWND pWndOwner, PMENU Menu,
       {
           PWND pWnd;
           if (!VerifyMenu(Item->spSubMenu)) return;
-          pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
           MENU_HideSubPopups(pWndOwner, Item->spSubMenu, FALSE, wFlags);
           MENU_SelectItem(pWndOwner, Item->spSubMenu, NO_SELECTED_ITEM, SendMenuSelect, NULL);
           TRACE("M_HSP top p hm %p  pWndOwner IDMenu %p\n",top_popup_hmenu,pWndOwner->IDMenu);
-          co_UserDestroyWindow(pWnd);
+          if (Item->spSubMenu->hWnd != NULL)
+          {
+              pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
+              if (pWnd != NULL)
+              {
+                  co_UserDestroyWindow(pWnd);
+              }
+          }
```

Comment:
The first `if (Item->spSubMenu->hWnd != NULL)` check is redundant because ValidateHwndNoErr does it (it's part of validating the hwnd).
```suggestion
          pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
          if (pWnd)
              co_UserDestroyWindow(pWnd);
```